astropy/visualization/wcsaxes/tests/test_wcsapi.py

```diff
@@ -217,6 +219,86 @@ def test_coord_type_1d_2d_wcs_uncorrelated():
+    CTYPE4  = 'HPLN-TAN'           / Coordinate type codegnomonic projection
+    CRVAL1  =                  0.0 / [min] Coordinate value at reference point
+    CRVAL2  =                  0.0 / [m] Coordinate value at reference point
+    CRVAL3  =                  0.0 / [deg] Coordinate value at reference point
+    CRVAL4  =                  0.0 / [deg] Coordinate value at reference point
+    LONPOLE =                180.0 / [deg] Native longitude of celestial pole
+    LATPOLE =                  0.0 / [deg] Native latitude of celestial pole
+    """)
+    return WCS(header=header)
+
+
+@pytest.fixture
+def sub_wcs(wcs_4d, wcs_slice):
+    return SlicedLowLevelWCS(wcs_4d, wcs_slice)
+
+
+@pytest.mark.parametrize(("wcs_slice", "wcsaxes_slices", "world_map", "ndim"),
+                         [
+                             (np.s_[...], [0,0,'x','y'], (2, 3), 2),
+                             (np.s_[...], [0,'x',0,'y'], (1, 3), 3),
+                             (np.s_[...], ['x',0,0,'y'], (0, 3), 3),
+                             (np.s_[...], ['x','y',0,0], (0, 1), 2),
+                             (np.s_[:,:,0,:], [0, 'x', 'y'], (1, 2), 2),
+                             (np.s_[:,:,0,:], ['x', 0, 'y'], (0, 2), 3),
+                             (np.s_[:,:,0,:], ['x', 'y', 0], (0, 1), 3),
+                             (np.s_[:,0,:,:], ['x', 'y', 0], (0, 1), 2),
+                         ])
+def test_apply_slices(sub_wcs, wcs_slice, wcsaxes_slices, world_map, ndim):
+    transform_wcs, _, world_map = apply_slices(sub_wcs, wcsaxes_slices)
+    assert transform_wcs.world_n_dim == ndim
+
+    assert world_map == world_map
+
+
+# parametrize here to pass to the fixture
+@pytest.mark.parametrize("wcs_slice", [np.s_[:,:,0,:]])
+def test_sliced_ND_input(sub_wcs, wcs_slice):
+    slices_wcsaxes = [0, 'x', 'y']
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings('ignore', category=FutureWarning)
+        _, coord_meta = transform_coord_meta_from_wcs(sub_wcs, RectangularFrame, slices=slices_wcsaxes)
+
+    assert all(len(x) == 3 for x in coord_meta.values())
+
+    coord_meta['name'] = ['time', 'custom:pos.helioprojective.lat', 'custom:pos.helioprojective.lon']
+    coord_meta['type'] = ['scalar', 'latitude', 'longitude']
+    coord_meta['wrap'] = [None, None, 180.0]
+    coord_meta['unit'] = [u.Unit("min"), u.Unit("deg"), u.Unit("deg")]
+    coord_meta['visible'] = [False, True, True]
+    coord_meta['format_unit'] = [u.Unit("min"), u.Unit("arcsec"), u.Unit("arcsec")]
+    coord_meta['default_axislabel_position'] = ['', 'b', 't']
+    coord_meta['default_ticklabel_position'] = ['', 'b', 't']
+    coord_meta['default_ticks_position'] = ['', 'btlr', 'btlr']
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings('ignore', category=FutureWarning)
+        # Validate the axes initialize correctly
+        ax = plt.subplot(projection=sub_wcs, slices=slices_wcsaxes)
```

Comment:
So is there anything to check here or are you just making sure the initialization doesn't crash?

Reply:
Just making sure the init doesn't crash.